pulsar-io/azure-data-explorer/src/main/java/org/apache/pulsar/io/azuredataexplorer/ADXSinkConfig.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.io.azuredataexplorer;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Objects;
+import lombok.Data;
+import lombok.experimental.Accessors;
+import org.apache.pulsar.io.core.annotations.FieldDoc;
+
+
+@Data
+@Accessors(chain = true)
+public class ADXSinkConfig implements Serializable {
+
+    @FieldDoc(required = true, defaultValue = "", help = "The ADX cluster URL")
+    private String clusterUrl;
+
+    @FieldDoc(required = true, defaultValue = "", help = "The database name to which data need to be ingested")
+    private String database;
+
+    @FieldDoc(required = true, defaultValue = "", help = "Table name to which pulsar data need to be ingested.")
+    private String table;
+
+    @FieldDoc(defaultValue = "", help = "The AAD app Id for authentication")
+    private String appId;
+
+    @FieldDoc(defaultValue = "", help = "The AAD app secret for authentication")
```

Comment:
```suggestion
    @FieldDoc(defaultValue = "", help = "The AAD app secret for authentication", sensitive = true)
```
secrets should be annotated with `sensitive = true` and loaded using the secrets API. The SinkContext has a method getSecret to get the value. (similar as https://pulsar.apache.org/docs/3.2.x/functions-develop-security/#step-2-get-the-secret for Pulsar Functions). This was fairly recently addressed for the existing connectors with PR https://github.com/apache/pulsar/pull/21675 .

Reply:
Sorry @lhotari  I missed this, will take care.